integration-tests/validatePIIRequest.spec.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  getLatestLogStreamName,
+  getMatchingLogEvents,
+  extractRequestID
+} from './utils/helpers'
+import { FilteredLogEvent } from '@aws-sdk/client-cloudwatch-logs'
+
+import { createZendeskRequest } from './utils/raiseZendeskRequest'
+import { approveZendeskRequest } from './utils/approveZendeskRequest'
+
+describe('Submit a PII request with approved ticket data', () => {
+  jest.setTimeout(45000)
+
+  it('Should log a success in cloud watch if Zendesk request is valid', async () => {
+    const ticketID = await createZendeskRequest(true)
+
+    await approveZendeskRequest(ticketID)
+
+    // Cloudwatch - Fetch latest log stream containing the ticket details
+    let latestLogStreamName = await getLatestLogStreamName()
+    console.log(`LATEST LOG STREAM NAME: ${latestLogStreamName}`)
+    let logStreamRequestID
+    let eventMatched = false
+
+    while (!eventMatched) {
+      const logEvents = await getMatchingLogEvents(
+        'INFO received Zendesk webhook',
+        latestLogStreamName
+      )
+      if (logEvents.length == 0) {
+        latestLogStreamName = await getLatestLogStreamName()
+        continue
+      }
+      const matchingEvent = logEvents.filter((event) => {
+        return (
+          event.message?.includes(`"zendeskId`) &&
+          event.message.includes(`"${ticketID}`)
+        )
+      })
+      if (matchingEvent.length == 1) {
+        console.log(
+          `Event for ticket ${ticketID} found in log stream: ${latestLogStreamName}`
+        )
+        const message: string = matchingEvent[0].message as string
+        console.log(`MATCHED EVENT: ${message}`)
+        logStreamRequestID = extractRequestID(message)
+        eventMatched = true
```

Comment:
I've not run the code, so not 100% sure, but couldn't you just break here?

Reply:
The loop would break there as there isn't any more code to run after that. I've added `break` for more clarity.